src/main/java/software/amazon/smithy/lsp/SmithyTextDocumentService.java

```diff
@@ -510,6 +535,34 @@ private List<ValidationEvent> getValidationEventsForShape(Shape shape) {
                 .collect(Collectors.toList());
     }
 
+    @Override
+    public CompletableFuture<List<Either<Command, CodeAction>>> codeAction(CodeActionParams params) {
```

Comment:
I wonder if we could take the implementation of this method and stick it somewhere in the new codeactions folder, maybe in SmithyCodeActions.java? That way, we can more easily test the code actions in isolation, and not add more bloat to the text document service class.